### PR TITLE
Eslint react perf

### PR DESCRIPTION
**Pull request by hanars**
re-enable eslint rules for react performance enhancements. Specifically, this prevents the creation of objects, arrays, and functions as props in the same scope as the component they are passed to. This improved react performance, as otherwise React will always see these props as changed and will re-render the component. 

For many of these change, it made sense to convert a functional component to a react class, so the diff is quite large despite not much actually changing

### CODE REVIEW

File: ui/shared/components/panel/family/FamilyLayout.jsx [hunk 1]
```diff
@@ -26,11 +26,11 @@ const FamilyLayout = React.memo((
     <FamilyGrid annotation={annotation} offset={offset}>
       <Grid.Row>
         {(leftContent || !useFullWidth) && <Grid.Column width={3}>{leftContent}</Grid.Column>}
-        {compact ? fields.map(
+        {compact ? (fields || []).map(
           field => <Grid.Column width={field.colWidth || 1} key={field.id}>{fieldDisplay(field)}</Grid.Column>,
         ) : (
           <Grid.Column width={getContentWidth(useFullWidth, leftContent, rightContent)}>
-            {fields.map(field => fieldDisplay(field))}
+            {(fields || []).map(field => fieldDisplay(field))}
```
Comment (ShifaSZ) [inline ui/shared/components/panel/family/FamilyLayout.jsx:33]: Are there any real use-cases that `fields` can be null?

File: ui/shared/components/panel/genes/GeneDetail.jsx [hunk 2]
```diff
@@ -35,11 +37,11 @@ const GeneSection = React.memo(({ details }) => (
...
   </CompactGrid>
-))
+)
 
-GeneSection.propTypes = {
-  details: PropTypes.arrayOf(PropTypes.object),
-}
+// GeneSection.propTypes = {
+//   details: PropTypes.arrayOf(PropTypes.object),
+// }
```
Comment (ShifaSZ) [inline ui/shared/components/panel/genes/GeneDetail.jsx:44]: Do we still need these lines?

**Review by ShifaSZ — CHANGES_REQUESTED** [review]
I just ran through quickly.

File: ui/pages/SummaryData/components/SavedVariants.jsx [hunk 3]
```diff
@@ -64,27 +65,14 @@ TAG_OPTIONS.push({
...
 
-const BaseSavedVariants = React.memo(({ loadAllProjectSavedVariants, geneDetail, ...props }) => {
-  const { params } = props.match
-  const { tag, gene } = params
-
-  const getUpdateTagUrl = selectedTag => `/summary_data/saved_variants/${selectedTag}${gene ? `/${gene}` : ''}`
-
-  const getGeneHref = selectedGene => `/summary_data/saved_variants/${tag || SHOW_ALL}/${selectedGene.key}`
+const getUpdateTagUrl =
+  (selectedTag, match) => `/summary_data/saved_variants/${selectedTag}${match.params.gene ? `/${match.params.gene}` : ''}`
```
[the remark this answers is not in the document]
Comment (hanars) [reply ui/pages/SummaryData/components/SavedVariants.jsx:69]: no, literally the entire purpose of this PR is to stop defining function props in the same scope that they are used

File: ui/shared/components/panel/genes/GeneDetail.jsx (hunk 2 above)
Comment (hanars) [reply ui/shared/components/panel/genes/GeneDetail.jsx:44]: no, good catch

**Pull request merged**